src/main/java/server/usecases/friendinteractors/requestfriend/requestFriendInteractor.java

```diff
@@ -15,41 +16,46 @@ public requestFriendInteractor(requestFriendOutputBoundary output, requestFriend
+        boolean b3 = dataBase.notAlreadyFriend(model.getRequesterid(), model.getFriendid());
+        boolean b4 = dataBase.notAlreadyRequested(model.getRequesterid(), model.getFriendid());
+        boolean b5 = model.getRequesterid() != model.getFriendid();
+        dataBase.cleanRequesterListDuplicateUID(model.getRequesterid());
+        if(b1 && b3 && b4 && b5){
             int requesterid = model.getRequesterid();
             int friendid = model.getFriendid();
-            String requesterName = dataBase.getUserName(requesterid);
-//            String requesterAddress = dataBase.getAddress(requesterid);
-//            int requesterPeerPort = dataBase.getPeerPort(requesterid);
+            String requesterName = dataBase.getUserNamebyUID(requesterid);
+            String requesterAddress = dataBase.getAddress(requesterid);
+            int requesterPeerPort = dataBase.getPeerPort(requesterid);
             String friendAddress = dataBase.getAddress(friendid);
             int friendPeerPort = dataBase.getPeerPort(friendid);
 
-            //output.success(requesterid, requesterAddress, requesterPeerPort);
+            dataBase.requestFriendbyID(requesterid, friendid);
+            output.success(requesterid, requesterAddress, requesterPeerPort);
             output.reportToFriend(requesterid, requesterName, friendid, friendAddress, friendPeerPort);
+            return;
         }
-        else
-        {
-            // input not found in database
-//            int requesterid = model.getRequesterid();
-//            String requesterAddress = dataBase.getAddress(requesterid);
-//            int requesterPeerPort = dataBase.getPeerPort(requesterid);
-//            output.fail(requesterid, requesterAddress, requesterPeerPort);
-        }
-        if(b2){
-//            String requesterName = model.getRequesterName();
-//            String friendName = model.getFriendName();
-//            int friendid = dataBase.getUserid(friendName);
-//            int requesterid = dataBase.getUserid(requesterName);
-//            String requesterAddress = dataBase.getAddress(requesterid);
-//            int requesterPeerPort = dataBase.getPeerPort(requesterid);
-//            String friendAddress = dataBase.getAddress(friendid);
-//            int friendPeerPort = dataBase.getPeerPort(friendid);
-//
-//            output.success(requesterid, requesterAddress, requesterPeerPort);
-//            output.reportToFriend(requesterid, requesterName, friendid, friendAddress, friendPeerPort);
+        if(b2 && b3 && b4 && b5){
+            String requesterName = model.getRequesterName();
+            String friendName = model.getFriendName();
+            int friendid = dataBase.getUIDbyUserName(friendName);
+            int requesterid = dataBase.getUIDbyUserName(requesterName);
+            String requesterAddress = dataBase.getAddress(requesterid);
+            int requesterPeerPort = dataBase.getPeerPort(requesterid);
```

Comment:
Im not sure if the users will ever request friend by their name, maybe comment it out?

Reply:
It works anyway